src/websocket/server.cc

```diff
@@ -177,23 +260,67 @@ future<> connection::read_loop() {
         });
     }).then_wrapped([this] (future<> f) {
         if (f.failed()) {
-            wlogger.error("Read failed: {}", f.get_exception());
+            wlogger.error("Failure: {}", f.get_exception());
         }
         return _replies.push_eventually({});
     }).finally([this] {
         return _read_buf.close();
     });
 }
 
+future<> connection::send_data(temporary_buffer<char> buff) {
+    sstring data;
+    data.append("\x81", 1);
+    if ((126 <= buff.size()) && (buff.size() <= std::numeric_limits<uint16_t>::max())) {
+        uint16_t length = buff.size();
+        length = htobe16(length);
+        data.append("\x7e", 1);
+        data.append(reinterpret_cast<char*>(&length), sizeof(uint16_t));
+    } else if (std::numeric_limits<uint16_t>::max() < buff.size()) {
+        uint64_t length = buff.size();
+        length = htobe64(length);
+        data.append("\x7f", 1);
+        data.append(reinterpret_cast<char*>(&length), sizeof(uint64_t));
+    } else {
+        uint8_t length = buff.size() & 0x7F;
+        data.append(reinterpret_cast<char*>(&length), sizeof(uint8_t));
+    }
+    data.append(buff.get(), buff.size()); 
+    temporary_buffer<char> result{data.data(), data.length()}; 
+    return _write_buf.write(std::move(result)).then([this]{
+        return _write_buf.flush();
+    }); 
+}
+
 future<> connection::response_loop() {
-    // FIXME: implement
-    return make_ready_future<>();
+    return do_until([this] {return _done;}, [this] {
+        return input().read().then([this](temporary_buffer<char> buf) {
+            // FIXME: implement
+            wlogger.info("Loop: {} {}", buf.get(), buf.size());
+            return send_data(std::move(buf));
+            //return this->_server.handlers["echo"](std::move(buf), _write_buf);
```

Comment:
Note about the commented-out line: instead of hardcoding "echo" here, remember that we want the handler to be picked dynamically, depending what subprotocol the user defined in the handshake. Ref: https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API/Writing_WebSocket_servers#subprotocols

Basically, the implementation should go like this:
 1. During the handshake, look into http headers for `Sec-WebSocket-Protocol`
 2. The subprotocol is looked up in the map of known handlers (the ones that somebody provided via `register_handler` when instantiating the server).
 3. If no matching handler is found, handshake should fail with an error saying that the subprotocol is not known
 4. If a handler was found, it should be remembered in the connection and called here.
 
Note that we should also allow users to register a "default" handler, which works if somebody does not send a `Sec-WebSocket-Protocol` header during handshake. That can be implemented later as a follow-up.